vistalink-models/src/main/java/gov/va/api/lighthouse/vistalink/models/vprgetpatientdata/VprGetPatientData.java

```
@@ -0,0 +1,98 @@
+package gov.va.api.lighthouse.vistalink.models.vprgetpatientdata;
+
+import gov.va.api.lighthouse.vistalink.api.RpcDetails;
+import gov.va.api.lighthouse.vistalink.api.RpcInvocationResult;
+import gov.va.api.lighthouse.vistalink.models.TypeSafeRpc;
+import gov.va.api.lighthouse.vistalink.models.TypeSafeRpcResponse;
+import gov.va.api.lighthouse.vistalink.models.XmlResponseRpc;
+import java.time.Instant;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.Builder;
+import lombok.Data;
+
+public class VprGetPatientData
+    implements TypeSafeRpc<VprGetPatientData.Request, VprGetPatientData.Response> {
+  private static final String RPC_NAME = "VPR GET PATIENT DATA";
+
+  private static final String RPC_CONTEXT = "VPR APPLICATION PROXY";
+
+  /** Serialize the RPC results to a response object. */
+  public VprGetPatientData.Response fromResult(List<RpcInvocationResult> results) {
+    // TO-DO implement with help of XmlResponseRpc utility class
+    return Response.builder()
+        .results(
+            results.stream()
+                .filter(r -> r.error().isEmpty())
```

Not sure how I feel. Should the model class know how to filter the errors, or is that more an expectation of the class that is managing the results received from the Vistalink API? Instead of a list of results, should this just expect a list of `result.response`'s?

---

We might end up seeing a change in the API as we start using it. Error handling is not 100% clear to me atm. The model should not assume a behavior, e.g. ignore errors or throw exceptions on errors. I do think this is the responsibility of the "controller". However, we might need to structure this library to allow "pluggable behavior" around error handling. All this said, for now, I'm ok with this very optimistic implementation, assuming the controller will inspect the results _prior_ to trying to convert them.